src/components/Executions/filters/useUserByFilterState.ts

```diff
@@ -0,0 +1,53 @@
+import { FilterOperationName } from 'models/AdminEntity/types';
+import { useEffect, useState, useContext } from 'react';
+import { FilterByUserState } from './types';
+import { useFilterButtonState } from './useFilterButtonState';
+import { APIContext } from 'components/data/apiContext';
+
+interface FilterByUserStateArgs {
+    queryStateKey: string;
+    filterOperation?: FilterOperationName;
+}
+
+/** Maintains the state for a button to be used for filtering by user.
+ */
+export function useUserByFilterState({
+    queryStateKey,
+    filterOperation = FilterOperationName.EQ
+}: FilterByUserStateArgs): FilterByUserState {
+    const { profile } = useContext(APIContext);
```

Comment:
What is the main purpose to use context here is to reduce api call.
As you said, we can use useUserProfile directly in every component, but also it will make a new api call every time.
So if we define profile in APIContext, we can also use profile in every component without making api call.

Reply:
`useUserProfile` has [caching set to true](https://github.com/flyteorg/flyteconsole/blob/master/src/components/hooks/useUserProfile.ts#L12) so you _shouldn't_ see multiple requests unless there is a bug in the useFetchable hook or something is causing it to force reload.
Are you seeing it issue multiple requests in the devtools?